src/main/java/io/kestra/storage/s3/S3Storage.java

```diff
@@ -183,20 +328,35 @@ public List<URI> deleteByPrefix(String tenantId, URI storagePrefix) throws IOExc
 
             return result.deleted().stream()
                 .map(DeletedObject::key)
-                .map(key -> createUri(key.replace(tenantId + "/", "")))
+                .map(k -> {
+                    if (k.endsWith("/")) {
+                        return k.substring(0, k.length() - 1);
+                    }
+                    return k;
+                })
```

Comment:
Using the ternary operator here will allow to use compact lambda form in one line